lib/Sema/TypeCheckDecl.cpp

```diff
@@ -914,8 +914,15 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
                     current->getFullName());
         tc.diagnose(other, diag::invalid_redecl_prev, other->getFullName());
       } else {
-        tc.diagnose(current, diag::invalid_redecl, current->getFullName());
-        tc.diagnose(other, diag::invalid_redecl_prev, other->getFullName());
+        const auto *otherInit = dyn_cast<ConstructorDecl>(other);
+        if (otherInit && otherInit->isImplicit()) {
```

Comment:
This isn't the only reason a constructor can be implicit; it might be inherited. Maybe check that it's not overriding anything?

Reply:
Good point.